bot/exts/moderation/stream.py

```diff
@@ -0,0 +1,139 @@
+        \u2003`s` - seconds
+
+        Alternatively, an ISO 8601 timestamp can be provided for the duration.
+        """
+        # if duration is none then calculate default duration
+        if duration is None:
+            now = datetime.datetime.utcnow()
+            duration = now + datetime.timedelta(minutes=VideoPermission.default_permission_duration)
+
+        # Check if user already has streaming permission
+        already_allowed = any(Roles.video == role.id for role in user.roles)
+        if already_allowed:
+            await ctx.send(f"{Emojis.cross_mark} This user can already stream.")
+            return
+
+        # Schedule task to remove streaming permission from Member and add it to task cache
+        self.scheduler.schedule_at(duration, user.id, self._remove_streaming_permission(user))
+        await self._add_to_redis_cache(user.id, duration.timestamp())
+        await user.add_roles(discord.Object(Roles.video), reason="Temporary streaming access granted")
+        duration = format_infraction_with_duration(str(duration))
+        await ctx.send(f"{Emojis.check_mark} {user.mention} can now stream until {duration}.")
+
+    @commands.command(aliases=("pstream",))
+    @commands.has_any_role(*STAFF_ROLES)
+    async def permanentstream(
+            self,
+            ctx: commands.Context,
+            user: discord.Member,
+            *_
+    ) -> None:
+        """Permanently give user a streaming permission."""
+        # Check if user already has streaming permission
+        already_allowed = any(Roles.video == role.id for role in user.roles)
+        if already_allowed:
+            if user.id in self.scheduler:
+                self.scheduler.cancel(user.id)
+                await self._delete_from_redis(user.id)
+                await ctx.send(f"{Emojis.check_mark} Moved temporary permission to permanent")
+                return
+            await ctx.send(f"{Emojis.cross_mark} This user can already stream.")
+            return
+
+        await user.add_roles(discord.Object(Roles.video), reason="Permanent streaming access granted")
+        await ctx.send(f"{Emojis.check_mark} {user.mention} can now stream forever")
+
+    @commands.command(aliases=("unstream", ))
+    @commands.has_any_role(*STAFF_ROLES)
+    async def revokestream(
+            self,
+            ctx: commands.Context,
+            user: discord.Member
+    ) -> None:
+        """Take away streaming permission from a user."""
+        # Check if user has the streaming permission to begin with
+        allowed = any(Roles.video == role.id for role in user.roles)
+        if allowed:
+            # Cancel scheduled task to take away streaming permission to avoid errors
+            if user.id in self.scheduler:
+                self.scheduler.cancel(user.id)
+            await self._remove_streaming_permission(user)
+            await ctx.send(f"{Emojis.check_mark} Streaming permission taken from {user.display_name}.")
```

Comment:
Revoked is a nicer word. Also, why does this use the display name rather than the mention? Whichever it is, it should be consistent.
```suggestion
            await ctx.send(f"{Emojis.check_mark} Revoked the permission to stream from {user.mention}.")
```